### PR TITLE
Caching section (Cache component + caching by tags)

### PR DESCRIPTION
**Pull request by Borales**

### CODE REVIEW

File: docs/falcon-server/caching.md [hunk 1]
```diff
@@ -2,11 +2,138 @@
...
-their own caching options. Such cache will be handled by `@cache` directive itself, so the Extension does not have
-to do anything by itself in terms of cache checks, cache-key generations, etc.
+Falcon-Server provides several ways of working with cache.
 
-## Schema Caching options
+It provides `cache` component through GQL context available for all GraphQL Query and Mutation resolver methods.
+
+It also exposes `@cache` directive through the Base GraphQL Schema to the rest of the extensions to define their own caching options.
+Such cache will be handled by `@cache` directive itself, so the Extension does not have to do anything by itself in terms of cache checks,
+cache-key generations, etc.
```
Comment (mateuszwozniak) [inline docs/falcon-server/caching.md]: unnecessary enter?

File: docs/falcon-server/caching.md [hunk 2]
```diff
@@ -2,11 +2,138 @@
...
+
+## Cache adapters
+
+By default, Falcon-Server uses
+[`InMemoryLRUCache`](https://www.apollographql.com/docs/apollo-server/features/data-sources#using-memcachedredis-as-a-cache-storage-backend)
+as a cache backend.
```
Comment (mateuszwozniak) [inline docs/falcon-server/caching.md]: Those line breaks are weird. I assume that's some IDE's auto formatting?

File: docs/falcon-server/caching.md [hunk 3]
```diff
@@ -2,11 +2,138 @@
...
+}
+```
+
+### `get` cache
+
+`async get(key: string, setOptions?: GetCacheOptions): Promise<CacheResult>` method accepts the following arguments:
+
+- `key: string` - cache key
+- `setOptions?: GetCacheOptions` - optional object with the following sub-keys:
+  - `fetchData: () => Promise<GetCacheFetchResult>` - this is an optional async fetch-callback, which should return a value if the cached value is missing
```
Comment (mateuszwozniak) [inline docs/falcon-server/caching.md]:
```suggestion
  - `fetchData: () => Promise<GetCacheFetchResult>` - this is an optional async fetch-callback, which should return a value if the cached value is missing. The fetched data will be then cached for a later use.
```

File: docs/falcon-server/caching.md [hunk 4]
```diff
@@ -2,11 +2,138 @@
...
+  - `options: SetCacheOptions` - options object to be passed to [`set`](#set-cache) method
+
+Example of using `keyOrOptions` as an object with `fetchData` callback. Here's a piece of code how you would normally work with the cache:
```
Comment (mateuszwozniak) [inline docs/falcon-server/caching.md]:
```suggestion
Example of using `keyOrOptions` as an object with `fetchData` callback. Here's a piece of code that shows how you would normally work with the cache:
```

File: docs/falcon-server/caching.md [hunk 5]
```diff
@@ -115,37 +242,84 @@ Cache options are being checked with this order (from highest to lowest priority
...
-```json
-{
-  "cache": {
-    "type": "redis",
-    "options": {
-      "host": "redis-server"
-    }
-  }
+Thus if you run `query { item(id: 1) { id name } }` GQL query, `@cache` will generate `Item` and `Item:1`
+tags automatically for you, and will put the resolver value to the cache with the generated tags for further validation.
```
Comment (mateuszwozniak) [inline docs/falcon-server/caching.md]:
```suggestion
tags automatically for you, and will put the resolved value into the cache with the generated tags for further validation.
```

File: docs/falcon-server/caching.md [hunk 6]
```diff
@@ -115,37 +242,84 @@ Cache options are being checked with this order (from highest to lowest priority
...
+}
+
+type Item {
+  id: ID!
+  name: String
 }
 ```
 
-<!--memcached-->
+Such `@cache(idPath: ["children"])` statement will tell to the `@cache` directive to look in `children` key
```
Comment (mateuszwozniak) [inline docs/falcon-server/caching.md]:
```suggestion
Such `@cache(idPath: ["children"])` statement will tell to the `@cache` directive to look in `children` property
```

File: docs/falcon-server/caching.md (hunk 1 above)
Comment (Borales) [reply docs/falcon-server/caching.md]: @mateuszwozniak I usually try to keep the line length within a visual part of the IDE (120-140 chars), but such "newlines" have no effects on the page output (markdown does not count it as a paragraph).

File: docs/falcon-server/caching.md [hunk 7]
```diff
@@ -2,11 +2,138 @@
...
+    "options": ["memcached-server-1", "memcached-server-2"]
+  }
+}
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Cache Component
+
+Falcon-Server provides Cache wrapper-component with additional features required for a proper work.
```
Comment (cane-lagerwaard) [inline docs/falcon-server/caching.md]: "for a proper work"?

File: docs/falcon-server/caching.md [hunk 8]
```diff
@@ -115,37 +242,84 @@ Cache options are being checked with this order (from highest to lowest priority
...
-    }
-  }
+Thus if you run `query { item(id: 1) { id name } }` GQL query, `@cache` will generate `Item` and `Item:1`
+tags automatically for you, and will put the resolved value into the cache with the generated tags for further validation.
+The same works with a simple list, when you run `query { items { id name } }` - the directive will detect
+`[Item]` list and will extract `ID` values from each entry and will generate the following tags -
+`Item`, `Item:1`, `Item:2` and so on.
+
+This works great with "plain" values. But when you're dealing with nested listings - you need to help `@cache`
+directive by pointing out where to look for `ID` values at. Let's assume you have the following GQL schema:
```
Comment (cane-lagerwaard) [inline docs/falcon-server/caching.md]:
```suggestion
directive by pointing out where to look for `ID` values. Let's assume you have the following GQL schema:
```

File: docs/falcon-server/caching.md (hunk 7 above)
Comment (Borales) [reply docs/falcon-server/caching.md]: changed

**Pull request merged**